components/director/internal/open_resource_discovery/validation.go

```diff
@@ -132,21 +161,55 @@ func validateAPIInput(api *model.APIDefinitionInput) error {
 		validation.Field(&api.Industry, validation.By(func(value interface{}) error {
 			return validateJSONArrayOfStrings(value, regexp.MustCompile(StringArrayElementRegex))
 		})),
-		validation.Field(&api.ResourceDefinitions, validation.Required),
+		validation.Field(&api.ResourceDefinitions, validation.Required, validation.By(func(value interface{}) error {
+			if value == nil {
+				return nil
+			}
+
+			jsonArr := api.ResourceDefinitions
```

Comment:
`jsonArr` is a bit misleading because the ResourceDefinitions are not represented as `json.RawMessage`.

`resourceDefinitions` is better from my point of view